src/mono/mono/metadata/icall.c

```diff
@@ -1795,6 +1799,48 @@ ves_icall_RuntimeTypeHandle_IsInstanceOfType (MonoQCallTypeHandle type_handle, M
 	return !MONO_HANDLE_IS_NULL (inst);
 }
 
+void
+ves_icall_RuntimeMethodHandle_ReboxToNullable (MonoObjectHandle obj, MonoQCallTypeHandle type_handle, MonoObjectHandleOnStack res_handle, MonoError *error)
+{
+	MonoType *type = type_handle.type;
+	MonoClass *klass = mono_class_from_mono_type_internal (type);
+
+	mono_class_init_checked (klass, error);
+	if (!is_ok (error)) {
+		HANDLE_ON_STACK_SET (res_handle, NULL);
+		return;
+	}
+
+	MonoObjectHandle res = mono_object_new_handle (klass, error);
```

Comment:
This could use the newer way of handles, i.e. using raw object references and MONO_HANDLE_PIN ().